src/providers/tenderly-simulation-provider.ts

```diff
@@ -464,6 +464,19 @@ export class TenderlySimulator extends Simulator {
               2
             )}.`
           );
+
+          if ((resp.result[2] as JsonRpcError).error.data) {
```

Comment:
better to check if `resp.result[2]` exists here

Reply:
aw it's actually covered in the outer if condition:

```
if (
          !resp ||
          !resp.result ||
          resp.result.length < 3 ||
          (resp.result[2] as JsonRpcError).error
        )
```

Reply:
that `if` condition can still enter if resp/resp.result/resp.result[2] is not present - so here you want to do something like 
`if (resp.result.length >= 3 && (resp.result[2] as JsonRpcError).error.data) {`

Reply:
made it more defensive